### ISSUE TITLE
mypy error: Unsupported operand type for unary - ("Array")

### ISSUE THREAD

**Issue by peterroelants** [google/jaxtyping#32]
Using mypy to typecheck the following piece of code:
```python
import jax.numpy as jnp
from jaxtyping import Array, Float

def softplus_inv(x: Float[Array, "x"]) -> Float[Array, "x"]:
    """Inverse to softplus"""
    return x + jnp.log(-jnp.expm1(-x))
```

Results in:
```console
error: Unsupported operand type for unary - ("Array")
```
Am I doing something wrong? Other parts of my code seem to pass the type checks ok. It just seems that this unary minus `-` seems to cause issues.

Tested on:
- Python 3.10
- jaxtyping 0.2.5
- jax 0.3.17
- mypy 0.971

**Comment by peterroelants** [comment]
Same issue with:
```python
import jax.numpy as jnp
from jaxtyping import Array, Float

def inverse(x: Float[Array, "x"]) -> Float[Array, "x"]:
    return -x
```

Which results in:
```console
error: Unsupported operand type for unary - ("Array")
```

**Comment by peterroelants** [comment]
Weirdly the following also gives a similar error. E.g.
```python
def fn(x: Float[Array, "x"]) -> Float[Array, "x"]:
    return x + x
```
results in:
```console
error: Unsupported left operand type for + ("Array")
```

While the following codes does not give any issues when typing with mypy.

```python
def fn1(x: Float[Array, "x"]) -> Float[Array, "x"]:
    return x + jnp.array([2], dtype=jnp.float32)

def fn2(x: Float[Array, "x"]) -> Float[Array, "x"]:
    return x + jnp.array([2], dtype=jnp.float32) + x
```

**Comment by patrick-kidger** [comment]
Fixed in #34 -- thanks for the report.

The issue was a recent addition that mocked the `jaxtyping.Array` when generating documentation. (To produce prettier docs.) But the static type checker didn't know which version of `Array` was being used. So it took a guess and guessed wrong!

**Comment by peterroelants** [comment]
Thanks, I can confirm it working in 0.2.7
